Improve information presented for jndi lookup failures

Is there a likely deployment descriptor problem or class not found exception that is responsible for the following?  Could we include the cause exception or at least log it?

Exception call stack from EE 11 Persistence TCK testing:

```
acc._AS_INSTALL set to /home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/..
acc.APPCPATH set to target/lib/arquillian-protocol-lib.jar:target/lib/tck-porting-lib.jar
VMARGS = null
"/usr/lib/jvm/java-17-openjdk/bin/java" -Dorg.glassfish.gmbal.no.multipleUpperBoundsException=true --add-opens=java.base/java.lang=ALL-UNNAMED -Dcom.sun.aas.installRoot="/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/.." -Djava.security.policy="/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/../lib/appclient/client.policy" -classpath /home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/../lib/gf-client.jar:. -Djava.system.class.loader=org.glassfish.appclient.client.acc.agent.ACCAgentClassLoader -Xshare:off -Djava.security.auth.login.config="/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/../lib/appclient/appclientlogin.conf"  "-Djdk.tls.client.enableSessionTicketExtension=false"  "-Djdk.tls.server.enableSessionTicketExtension=false"  "-Djava.security.policy=/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/lib/appclient/client.policy"  "-Dcts.tmp=/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/jakartaeetck/tmp"  "-Djava.security.auth.login.config=/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/lib/appclient/appclientlogin.conf"  "-Djava.protocol.handler.pkgs=javax.net.ssl"  "-Djavax.net.ssl.keyStore=/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/jakartaeetck/bin/certificates/clientcert.jks"  "-Djavax.net.ssl.keyStorePassword=changeit"  "-Djavax.net.ssl.trustStore=/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/domains/domain1/config/cacerts.jks"  "-Djavax.xml.parsers.SAXParserFactory=com.sun.org.apache.xerces.internal.jaxp.SAXParserFactoryImpl"  "-Djavax.xml.parsers.DocumentBuilderFactory=com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderFactoryImpl"  "-Djavax.xml.transform.TransformerFactory=com.sun.org.apache.xalan.internal.xsltc.trax.TransformerFactoryImpl"  "-Dorg.xml.sax.driver=com.sun.org.apache.xerces.internal.parsers.SAXParser"  "-Dorg.xml.sax.parser=org.xml.sax.helpers.XMLReaderAdapter"  "-Doracle.jdbc.J2EE13Compliant=true"  "-Doracle.jdbc.mapDateToTimestamp"  "-Dstartup.login=false"  "-Dauth.gui=false"  "-Dlog.file.location=/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/domains/domain1/logs"  "-Dri.log.file.location=/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/domains/domain1/logs"  "-DwebServerHost.2=localhost"  "-DwebServerPort.2=8080"  "-Ddeliverable.class=com.sun.ts.lib.deliverable.cts.CTSDeliverable"  -javaagent:"/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/../lib/gf-client.jar"=argsfile="/tmp/acc6150892951202217625.dat"  -classpath  /home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/../lib/gf-client.jar:/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/appclient/jpa_core_EntityGraph_vehicles_client.jar:/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/lib/arquillian-protocol-lib.jar:/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/lib/tck-porting-lib.jar  ee.jakarta.tck.persistence.core.EntityGraph.Client  "-p"  "/tmp/tstest.jte"  "-ap"  "/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/sql/derby/derby.dml.sql"  "-classname"  "ee.jakarta.tck.persistence.core.EntityGraph.ClientStateless3Test"  "-t"  "getEntityGraphStringTest"  "-vehicle"  "stateless3"
************************************************************
* props file set to "/tmp/smarlow-cts-props.txt"
************************************************************
10-22-2024 11:40:00:  TRACE: #######  Value of harness.socket.retry.count is "10"
10-22-2024 11:40:00:  TRACE: #######  Value of harness.log.port is "2000"
10-22-2024 11:40:00:  TRACE: #######  Actual bind value of harness.log.port is "2000"
10-22-2024 11:40:00:  TRACE: Check if called from within test process, inTestHarness= true
10-22-2024 11:40:00:  TRACE: in ServiceEETest.run(), this URL is:  file:/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/appclient/jpa_core_EntityGraph_vehicles_client.jar
10-22-2024 11:40:00:  TRACE: VehicleClient URL is:  file:/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/appclient/jpa_core_EntityGraph_vehicles_client.jar
10-22-2024 11:40:00:  TRACE: VehicleClient class check if is vehicle class =  no, is not com.sun.ts.tests.common.vehicle.VehicleClient class
10-22-2024 11:40:00:  TRACE: Using appclient vehicle so set is vehicle client to true
10-22-2024 11:40:00:  TRACE: in ServiceEETest.run() method
10-22-2024 11:40:00:  TRACE: Vehicle to be used for this test is:  stateless3
Oct 22, 2024 11:40:02 AM org.glassfish.enterprise.iiop.impl.GlassFishORBManager getORB
INFO: ORB initialization succeeded: com.sun.corba.ee.impl.orb.ORBImpl@78d73b1b
10-22-2024 11:40:02:  ERROR: Test failed.
10-22-2024 11:40:02:  ERROR: javax.naming.NamingException: Lookup failed for java:comp/env/ejb/Stateless3VehicleBean in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: No object bound for java:comp/env/ejb/Stateless3VehicleBean]
	at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:836)
	at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:257)
	at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
	at com.sun.ts.lib.util.TSNamingContext.lookup(TSNamingContext.java:70)
	at com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleRunner.run(Stateless3VehicleRunner.java:41)
	at com.sun.ts.lib.harness.ServiceEETest.run(ServiceEETest.java:128)
	at com.sun.ts.lib.harness.EETest.getPropsReady(EETest.java:494)
	at com.sun.ts.lib.harness.ServiceEETest.run(ServiceEETest.java:232)
	at com.sun.ts.lib.harness.EETest.run(EETest.java:296)
	at ee.jakarta.tck.persistence.core.EntityGraph.Client.main(Client.java:48)
Caused by: javax.naming.NameNotFoundException: No object bound for java:comp/env/ejb/Stateless3VehicleBean
STATUS:Failed.Test run in stateless3 vehicle failed.
	at com.sun.enterprise.naming.impl.JavaURLContext.lookup(JavaURLContext.java:135)
	at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:818)
	... 9 more
	Suppressed: javax.naming.NamingException: Exception resolving Ejb for 'Remote ejb-ref name=ejb/Stateless3VehicleBean,Remote 3.x interface =com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF,ejb-link=Stateless3VehicleBean,lookup=,mappedName=,jndi-name=,refType=Session' .  Actual (possibly internal) Remote JNDI name used for lookup is '#com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF' [Root exception is javax.naming.NamingException: Lookup failed for #com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: #com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF not found]]
		at com.sun.ejb.EjbNamingReferenceManagerImpl.resolveEjbReference(EjbNamingReferenceManagerImpl.java:159)
		at com.sun.enterprise.container.common.impl.ComponentEnvManagerImpl$EjbReferenceProxy.create(ComponentEnvManagerImpl.java:975)
		at com.sun.enterprise.naming.impl.GlassfishNamingManagerImpl.lookup(GlassfishNamingManagerImpl.java:621)
		at com.sun.enterprise.naming.impl.GlassfishNamingManagerImpl.lookup(GlassfishNamingManagerImpl.java:587)
		at com.sun.enterprise.naming.impl.JavaURLContext.lambda$lookup$0(JavaURLContext.java:153)
		at com.sun.enterprise.naming.impl.JavaURLContext.lookupOrCollectException(JavaURLContext.java:550)
		at com.sun.enterprise.naming.impl.JavaURLContext.lookup(JavaURLContext.java:153)
		... 10 more
	Caused by: javax.naming.NamingException: Lookup failed for #com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: #com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF not found]
		at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:836)
		at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:257)
		at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
		at com.sun.ejb.EjbNamingReferenceManagerImpl.resolveEjbReference(EjbNamingReferenceManagerImpl.java:154)
		... 16 more
	Caused by: javax.naming.NameNotFoundException: #com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF not found
		at com.sun.enterprise.naming.impl.TransientContext.doLookup(TransientContext.java:234)
		at com.sun.enterprise.naming.impl.TransientContext.lookup(TransientContext.java:202)
		at com.sun.enterprise.naming.impl.SerialContextProviderImpl.lookup(SerialContextProviderImpl.java:37)
		at com.sun.enterprise.naming.impl.RemoteSerialContextProviderImpl.lookup(RemoteSerialContextProviderImpl.java:85)
		at jdk.internal.reflect.GeneratedMethodAccessor59.invoke(Unknown Source)
		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.base/java.lang.reflect.Method.invoke(Method.java:569)
		at com.sun.corba.ee.impl.presentation.rmi.ReflectiveTie.dispatchToMethod(ReflectiveTie.java:122)
		at com.sun.corba.ee.impl.presentation.rmi.ReflectiveTie._invoke(ReflectiveTie.java:152)
		at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatchToServant(ServerRequestDispatcherImpl.java:501)
		at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatch(ServerRequestDispatcherImpl.java:172)
		at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequestRequest(MessageMediatorImpl.java:1529)
		at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequest(MessageMediatorImpl.java:1405)
		at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleInput(MessageMediatorImpl.java:910)
		at com.sun.corba.ee.impl.protocol.giopmsgheaders.RequestMessage_1_2.callback(RequestMessage_1_2.java:192)
		at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequest(MessageMediatorImpl.java:674)
		at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.dispatch(MessageMediatorImpl.java:476)
		at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.doWork(MessageMediatorImpl.java:2202)
		at com.sun.corba.ee.impl.threadpool.ThreadPoolImpl$WorkerThread.performWork(ThreadPoolImpl.java:476)
		at com.sun.corba.ee.impl.threadpool.ThreadPoolImpl$WorkerThread.run(ThreadPoolImpl.java:519)

10-22-2024 11:40:02:  TRACE: SLEPT FOR:  0
```

App client container side output inclues:

```
INFO: [APPCLIENT-out] 10-22-2024 11:36:36:  TRACE: VehicleClient URL is:  file:/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/appclient/jpa_core_EntityGraph_vehicles_client.jar
Oct 22, 2024 11:36:36 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 10-22-2024 11:36:36:  TRACE: VehicleClient class check if is vehicle class =  no, is not com.sun.ts.tests.common.vehicle.VehicleClient class
Oct 22, 2024 11:36:36 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 10-22-2024 11:36:36:  TRACE: Using appclient vehicle so set is vehicle client to true
Oct 22, 2024 11:36:36 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 10-22-2024 11:36:36:  TRACE: in ServiceEETest.run() method
Oct 22, 2024 11:36:36 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 10-22-2024 11:36:36:  TRACE: Vehicle to be used for this test is:  appmanagedNoTx
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] Oct 22, 2024 11:36:38 AM org.glassfish.enterprise.iiop.impl.GlassFishORBManager getORB
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] INFO: ORB initialization succeeded: com.sun.corba.ee.impl.orb.ORBImpl@78d73b1b
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 10-22-2024 11:36:38:  ERROR: Test failed.
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 10-22-2024 11:36:38:  ERROR: javax.naming.NamingException: Lookup failed for java:comp/env/ejb/AppManagedNoTxVehicleBean in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: No object bound for java:comp/env/ejb/AppManagedNoTxVehicleBean]
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 	at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:836)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 	at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:257)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] STATUS:Failed.Test run in application-managed resource-local vehicle failed.
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 	at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 	at com.sun.ts.lib.util.TSNamingContext.lookup(TSNamingContext.java:70)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 	at com.sun.ts.tests.common.vehicle.appmanagedNoTx.AppManagedNoTxVehicleRunner.run(AppManagedNoTxVehicleRunner.java:38)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 	at com.sun.ts.lib.harness.ServiceEETest.run(ServiceEETest.java:128)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 	at com.sun.ts.lib.harness.EETest.getPropsReady(EETest.java:494)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 	at com.sun.ts.lib.harness.ServiceEETest.run(ServiceEETest.java:232)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 	at com.sun.ts.lib.harness.EETest.run(EETest.java:296)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 	at ee.jakarta.tck.persistence.core.EntityGraph.Client.main(Client.java:48)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] Caused by: javax.naming.NameNotFoundException: No object bound for java:comp/env/ejb/AppManagedNoTxVehicleBean
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 	at com.sun.enterprise.naming.impl.JavaURLContext.lookup(JavaURLContext.java:135)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 	at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:818)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 	... 9 more
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 	Suppressed: javax.naming.NamingException: Exception resolving Ejb for 'Remote ejb-ref name=ejb/AppManagedNoTxVehicleBean,Remote 3.x interface =com.sun.ts.tests.common.vehicle.appmanagedNoTx.AppManagedNoTxVehicleIF,ejb-link=AppManagedNoTxVehicleBean,lookup=,mappedName=,jndi-name=,refType=Session' .  Actual (possibly internal) Remote JNDI name used for lookup is '#com.sun.ts.tests.common.vehicle.appmanagedNoTx.AppManagedNoTxVehicleIF' [Root exception is javax.naming.NamingException: Lookup failed for #com.sun.ts.tests.common.vehicle.appmanagedNoTx.AppManagedNoTxVehicleIF in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: #com.sun.ts.tests.common.vehicle.appmanagedNoTx.AppManagedNoTxVehicleIF not found]]
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 		at com.sun.ejb.EjbNamingReferenceManagerImpl.resolveEjbReference(EjbNamingReferenceManagerImpl.java:159)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 		at com.sun.enterprise.container.common.impl.ComponentEnvManagerImpl$EjbReferenceProxy.create(ComponentEnvManagerImpl.java:975)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 		at com.sun.enterprise.naming.impl.GlassfishNamingManagerImpl.lookup(GlassfishNamingManagerImpl.java:621)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 		at com.sun.enterprise.naming.impl.GlassfishNamingManagerImpl.lookup(GlassfishNamingManagerImpl.java:587)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 		at com.sun.enterprise.naming.impl.JavaURLContext.lambda$lookup$0(JavaURLContext.java:153)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 		at com.sun.enterprise.naming.impl.JavaURLContext.lookupOrCollectException(JavaURLContext.java:550)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 		at com.sun.enterprise.naming.impl.JavaURLContext.lookup(JavaURLContext.java:153)
Oct 22, 2024 11:36:38 AM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
INFO: [APPCLIENT-out] 		... 10 more

```

Paste of conversation from Jakarta EE TCK Slack channel https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729612177243759:

> Scott Marlow
>   Oct 22nd at 11:49 AM
> Persistence TCK test (appclient) failure that is unclear to me as to the cause is included in https://github.com/eclipse-ee4j/glassfish/issues/25190
> 41 replies
> 
> 
> Scott Marlow
>   Oct 22nd at 11:50 AM
> Error is: No object bound for java:comp/env/ejb/Stateless3VehicleBean STATUS:Failed.Test run in stateless3 vehicle failed. but not the cause unless I missed it.
> 
> 
> Arjan Tijms
>   [Oct 22nd at 12:45 PM](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729615538909809?thread_ts=1729612177.243759&cid=C0131MLD538)
> Is it an option to attach a debugger to discover what might be going wrong?
> 
> 
> Scott Marlow
>   Oct 22nd at 12:50 PM
> I guess not at the moment since I couldn't seem to when I tried last week.
> 
> 
> Scott Marlow
>   [Oct 22nd at 12:57 PM](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729616259369039?thread_ts=1729612177.243759&cid=C0131MLD538)
> Is it possible that it is caused by a ClassNotFoundException?
> 
> 
> Arjan Tijms
>   Oct 23rd at 7:42 AM
>  I guess not at the moment since I couldn’t seem to when I tried last week.
> You should normally be able to attach a debugger easily by supplying the -Dglassfish.suspend parameter on the commandline when starting the maven build.
> E.g. mvn clean install -Dtest=…. -Dglassfish.suspend
> 
> 
> Brian Decker
>   [Oct 23rd at 9:47 AM](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729691245285999?thread_ts=1729612177.243759&cid=C0131MLD538)
> Coming from a position of not really knowing anything about Glassfish...
> Where is it getting the thing it's NNFEing on? From a binding for ejb/Stateless3VehicleBean?
> javax.naming.NameNotFoundException: #com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF not found
> That seems like an incomplete name somehow compared to the locations where that bean can actually be found on the server:
> Portable JNDI names for EJB Stateless3VehicleBean: [java:global/jpa_core_EntityGraph_vehicles/jpa_core_EntityGraph_stateless3_vehicle_ejb/Stateless3VehicleBean!com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF, java:global/jpa_core_EntityGraph_vehicles/jpa_core_EntityGraph_stateless3_vehicle_ejb/Stateless3VehicleBean]]]
> 
> Glassfish-specific (Non-portable) JNDI names for EJB Stateless3VehicleBean: [com_sun_ts_tests_common_vehicle_Stateless3VehicleBean#com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF, com_sun_ts_tests_common_vehicle_Stateless3VehicleBean]]]
> 
> 
> Scott Marlow
>   [Oct 23rd at 10:15 AM](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729692909540239?thread_ts=1729612177.243759&cid=C0131MLD538)
> [https://github.com/scottmarlow/jakartaee-tck/blob/correct_jpa_sunappclientxml/jpa/[…]arta/tck/persistence/core/EntityGraph/ClientStateless3Test.java](https://github.com/scottmarlow/jakartaee-tck/blob/correct_jpa_sunappclientxml/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/EntityGraph/ClientStateless3Test.java) adds com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF.class to the app client jar and ejb jar but maybe some other jar is missed but we don't see a CNFE so who knows.
> [10:15](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729692938876999?thread_ts=1729612177.243759&cid=C0131MLD538)
> But yeah, maybe its missing something else.
> 
> 
> Brian Decker
>   Oct 23rd at 10:17 AM
> It's looking up java:comp/env/ejb/Stateless3VehicleBean , which would be a pointer to a name in actual JNDI. It's that mapping I'm asking about.
> 
> 
> Scott Marlow
>   [Oct 23rd at 10:20 AM](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729693218378669?thread_ts=1729612177.243759&cid=C0131MLD538)
> I added the GF server.log output to https://github.com/eclipse-ee4j/glassfish/issues/25190 as a comment if that is helpful.  The java:comp/env/ejb/Stateless3VehicleBean lookup must be done from an actual EE component which I think app client container is considered to be as JNDI java:comp/env lookups will fail if not done from component.
> 
> 
> Brian Decker
>   [Oct 23rd at 10:21 AM](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729693263182329?thread_ts=1729612177.243759&cid=C0131MLD538)
> Yeah, I saw the server log. The part where I pasted the actual JNDI names being bound into naming is from that.
> 
> 
> Scott Marlow
>   Oct 23rd at 10:22 AM
> I'm mostly looking for an easy way to identify the problem so we can quickly address it and the many other ones likely similar to it.  We expect CNFEs + deployment descriptor issues but need better error information.
> [10:23](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729693385029369?thread_ts=1729612177.243759&cid=C0131MLD538)
> I'm not against using mvn clean install -Dtest=…. -Dglassfish.suspend if that helps us find the information quickly as well.
> 
> 
> Brian Decker
>   Oct 23rd at 10:27 AM
> I'm trying to make the argument that all of the information you need is already provided, but I can't be sure of that if we don't actually successfully determine the problem.
> The missing piece is knowing what the app config is that connects the java:comp name to the name in JNDI.
> 
> 
> Scott Marlow
>   [Oct 23rd at 10:30 AM](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729693852663669?thread_ts=1729612177.243759&cid=C0131MLD538)
> The alternative is to spend time trying to use the EE 10 JPA appclient tests instead which do work with GlassFish 7 but I'm not sure if they work with GlassFish 8.
> [@Arjan Tijms](https://eclipsefoundationhq.slack.com/team/UTHQNAMEV)
>  do you know if GlassFish 8 passes all of the EE 10 Platform TCK tests?  I know that our TCK CI test that we talked about didn't complete due to the job failing since the check for glassfish7 fails when the folder is called glassfish8 but perhaps you have seen results elsewhere?
> 
> 
> Scott Marlow
>   Friday at 10:24 AM
> I didn't have a chance to debug yet but wanted to report that -Dglassfish.suspend did seem to lead to :
> Starting container using command: [/usr/lib/jvm/java-17-openjdk/bin/java, -jar, /home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/modules/admin-cli.jar, start-domain, --suspend, -t]
> Debugging is enabled and the server is suspended. Please attach to the debugging port at: 9,009
> 
> 
> Arjan Tijms
>   [Friday at 1:01 PM](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729875664219429?thread_ts=1729612177.243759&cid=C0131MLD538)
> Yes, that should work indeed. Good to hear it also works for you :wink:
> 
> 
> Scott Marlow
>   [Friday at 2:38 PM](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729881535521559?thread_ts=1729612177.243759&cid=C0131MLD538)
> We do reach org.glassfish.ejb.startup.EjbDeployer.load()
> [2:39](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729881571049339?thread_ts=1729612177.243759&cid=C0131MLD538)
> So, where should I set a breakpoint on GF server side to get more info?
> 
> 
> Scott Marlow
>   Friday at 2:46 PM
> I seem to reach the server side InterceptorManager.intercept for POST_CONSTRUCT
> [2:47](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729882021264299?thread_ts=1729612177.243759&cid=C0131MLD538)
> For AppManagedVehicleBean; id: 90c02300a81f-ffffffffc4feef3b-0
> [2:47](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729882074260609?thread_ts=1729612177.243759&cid=C0131MLD538)
> now in callstack:
> getEJB:223, EJBContextImpl (com.sun.ejb.containers)
> 2 hidden frames
> aroundConstruct:60, CDIAroundConstructCallback (org.glassfish.weld.services)
> 7 hidden frames
> _createCDIInjectionContext:249, CDIServiceImpl (org.glassfish.weld.services)
> createCDIInjectionContext:179, CDIServiceImpl (org.glassfish.weld.services)
> 23 hidden frames
> [2:49](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729882155805979?thread_ts=1729612177.243759&cid=C0131MLD538)
> Now hitting post constructor for AppManagedVehicleBean; id: 90c02300a81f-ffffffffc4feef3b-2
> [2:49](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729882176598749?thread_ts=1729612177.243759&cid=C0131MLD538)
> ^ sounds like it is reaching the server during jndi lookup of bean
> [2:51](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729882278571659?thread_ts=1729612177.243759&cid=C0131MLD538)
> java.lang.ClassNotFoundException: org.glassfish.jndi.cosnaming.CNCtxFactory not found by org.glassfish.main.common.glassfish-naming [108]
> [2:51](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729882290111689?thread_ts=1729612177.243759&cid=C0131MLD538)
> I'm just catching all exceptions to see what is hit
> [2:53](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729882398275899?thread_ts=1729612177.243759&cid=C0131MLD538)
> cnfe for com.sun.ejb.codegen.GenericEJBHome_Generated
> [2:54](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729882452006609?thread_ts=1729612177.243759&cid=C0131MLD538)
> And com.sun.ejb.codegen.GenericEJBHome_Generated again, problem a lot of different classes being searched for
> [2:55](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729882511293729?thread_ts=1729612177.243759&cid=C0131MLD538)
> org.omg.CORBA.BAD_PARAM:   vmcid: 0x0  minor code: 0  completed: No
> [2:55](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729882523890829?thread_ts=1729612177.243759&cid=C0131MLD538)
> ^ for IORImpl[type=RMI:com.sun.ejb.codegen.GenericEJBHome_Generated:0000000000000000 iorTemplates=[IORTemplate[oktemp=ObjectKeyTemplate[magic=afabcb00 scid=38 serverid=100 orbid=S1AS-ORB oaid=ObjectAdapterID[RootPOA/#RFMBase#/113369499788836864_RBusiness_com.sun.ts.tests.common.vehicle.appmanaged._AppManagedVehicleIF_Remote-EJBHome] profile templates:[IIOPProfileTemplateImpl[giopVersion=1.2 primary=192.168.0.35:3700]]]]
> [2:56](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729882572616649?thread_ts=1729612177.243759&cid=C0131MLD538)
> cnfe for org.glassfish.jndi.cosnaming.CNCtxFactory
> [2:57](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729882644192279?thread_ts=1729612177.243759&cid=C0131MLD538)
> another corba bad param:
> IORImpl[type=RMI:com.sun.ejb.codegen.GenericEJBHome_Generated:0000000000000000 iorTemplates=[IORTemplate[oktemp=ObjectKeyTemplate[magic=afabcb00 scid=38 serverid=100 orbid=S1AS-ORB oaid=ObjectAdapterID[RootPOA/#RFMBase#/113369499788836864_RBusiness_com.sun.ts.tests.common.vehicle.appmanaged._AppManagedVehicleIF_Remote-EJBHome] profile templates:[IIOPProfileTemplateImpl[giopVersion=1.2 primary=192.168.0.35:3700]]]]
> [2:58](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729882737611729?thread_ts=1729612177.243759&cid=C0131MLD538)
> Now called into:
> intercept:310, InterceptorManager (com.sun.ejb.containers.interceptors)
> 1 hidden frame
> aroundConstruct:60, CDIAroundConstructCallback (org.glassfish.weld.services)
> 7 hidden frames
> _createCDIInjectionContext:249, CDIServiceImpl (org.glassfish.weld.services)
> createCDIInjectionContext:179, CDIServiceImpl (org.glassfish.weld.services)
> 23 hidden frames
> for ##########################################################
> InterceptorManager<com.sun.ts.tests.common.vehicle.appmanaged.AppManagedVehicleBean> has 1 interceptors
> 	beanClassName: com.sun.ts.tests.common.vehicle.appmanaged.AppManagedVehicleBean
> 	Interceptors:
> 		org.jboss.weld.module.ejb.SessionBeanInterceptor
> 	Callback Interceptors:
> 	0: interface jakarta.interceptor.AroundConstruct
> 		CallbackInterceptorChainImpl
> 	1: interface jakarta.annotation.PostConstruct
> 		CallbackInterceptorChainImpl
> 		callback[0]: public java.lang.Object org.jboss.weld.module.ejb.SessionBeanInterceptor.aroundInvoke(jakarta.interceptor.InvocationContext) throws java.lang.Exception
> 	2: interface jakarta.annotation.PreDestroy
> 		CallbackInterceptorChainImpl
> 	3: interface jakarta.ejb.PrePassivate
> 		CallbackInterceptorChainImpl
> 	4: interface jakarta.ejb.PostActivate
> 		CallbackInterceptorChainImpl
> ##########################################################
> [3:00](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729882845229079?thread_ts=1729612177.243759&cid=C0131MLD538)
> Next is failure:
> this = {[com.sun.corba.ee](http://com.sun.corba.ee/).impl.transport.ConnectionImpl@16843} "SocketOrChannelConnectionImpl[ java.nio.channels.SocketChannel[connected local=/192.168.0.35:3700 remote=/192.168.0.35:39580] ESTABLISHED true true]"
> this = {[com.sun.corba.ee](http://com.sun.corba.ee/).impl.transport.ConnectionImpl@16843} "SocketOrChannelConnectionImpl[ java.nio.channels.SocketChannel[connected local=/192.168.0.35:3700 remote=/192.168.0.35:39580] ESTABLISHED true true]"
> Exception = {[java.io](http://java.io/).IOException@16841}
>  backtrace = {java.lang.Object[6]@16848}
>  detailMessage = "End-of-stream"
>  cause = {[java.io](http://java.io/).IOException@16841} "java.io.IOException: End-of-stream"
>  stackTrace = {java.lang.StackTraceElement[5]@16851}
>  depth = 5
>  suppressedExceptions = {java.util.Collections$EmptyList@16684}  size = 0
> [3:01](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729882875299889?thread_ts=1729612177.243759&cid=C0131MLD538)
> FINE: 00410011: IOException received when reading from connection SocketOrChannelConnectionImpl[ java.nio.channels.SocketChannel[connected local=/192.168.0.35:3700 remote=/192.168.0.35:39580] ESTABLISHED true true]
> [3:02](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729882944164139?thread_ts=1729612177.243759&cid=C0131MLD538)
> appclient side might not appreciate me stepping through code a line at a time, but I also wonder what the above tells us?  Did we get further or the same?
> [3:03](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729883019185919?thread_ts=1729612177.243759&cid=C0131MLD538)
> app client side output for ^:
> INFO: [APPCLIENT-err] SEVERE: Class [ ee/jakarta/tck/persistence/core/StoredProcedureQuery/EmployeeMappedSC ] not found. Error while loading [ class ee.jakarta.tck.persistence.core.StoredProcedureQuery.Client1 ]
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
> INFO: [APPCLIENT-out] ************************************************************
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
> INFO: [APPCLIENT-out] * props file set to "/tmp/smarlow-cts-props.txt"
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
> INFO: [APPCLIENT-out] ************************************************************
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
> INFO: [APPCLIENT-out] 10-25-2024 15:00:08:  TRACE: #######  Value of harness.socket.retry.count is "10"
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
> INFO: [APPCLIENT-out] 10-25-2024 15:00:08:  TRACE: #######  Value of harness.log.port is "2000"
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
> INFO: [APPCLIENT-out] 10-25-2024 15:00:08:  TRACE: #######  Actual bind value of harness.log.port is "2000"
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
> INFO: [APPCLIENT-out] 10-25-2024 15:00:08:  TRACE: Check if called from within test process, inTestHarness= true
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
> INFO: [APPCLIENT-out] 10-25-2024 15:00:08:  TRACE: in ServiceEETest.run(), this URL is:  file:/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/appclient/jpa_core_StoredProcedureQuery_vehicles_client.jar
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
> INFO: [APPCLIENT-out] 10-25-2024 15:00:08:  TRACE: VehicleClient URL is:  file:/home/smarlow/tck/platformtck/glassfish-runner/jpa-platform-tck/target/appclient/jpa_core_StoredProcedureQuery_vehicles_client.jar
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
> INFO: [APPCLIENT-out] 10-25-2024 15:00:08:  TRACE: VehicleClient class check if is vehicle class =  no, is not com.sun.ts.tests.common.vehicle.VehicleClient class
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
> INFO: [APPCLIENT-out] 10-25-2024 15:00:08:  TRACE: Using appclient vehicle so set is vehicle client to true
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
> INFO: [APPCLIENT-out] 10-25-2024 15:00:08:  TRACE: in ServiceEETest.run() method
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd outputLineReceived
> INFO: [APPCLIENT-out] 10-25-2024 15:00:08:  TRACE: Vehicle to be used for this test is:  appmanaged
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err] javax.naming.NoInitialContextException: Cannot instantiate class: foo [Root exception is java.lang.ClassNotFoundException: foo]
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at java.naming/javax.naming.spi.NamingManager.getFactory(NamingManager.java:762)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at java.naming/javax.naming.spi.NamingManager.lambda$getInitialContext$1(NamingManager.java:724)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at java.base/jdk.internal.loader.AbstractClassLoaderValue$Memoizer.get(AbstractClassLoaderValue.java:329)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at java.base/jdk.internal.loader.AbstractClassLoaderValue.computeIfAbsent(AbstractClassLoaderValue.java:205)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at java.naming/javax.naming.spi.NamingManager.getInitialContext(NamingManager.java:724)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at java.naming/javax.naming.InitialContext.getDefaultInitCtx(InitialContext.java:305)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at java.naming/javax.naming.InitialContext.init(InitialContext.java:236)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at java.naming/javax.naming.InitialContext.<init>(InitialContext.java:208)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at com.sun.ts.lib.util.TSNamingContext.lookup(TSNamingContext.java:68)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at com.sun.ts.tests.common.vehicle.appmanaged.AppManagedVehicleRunner.run(AppManagedVehicleRunner.java:45)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at com.sun.ts.lib.harness.ServiceEETest.run(ServiceEETest.java:128)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at com.sun.ts.lib.harness.EETest.getPropsReady(EETest.java:494)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at com.sun.ts.lib.harness.ServiceEETest.run(ServiceEETest.java:232)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at com.sun.ts.lib.harness.EETest.run(EETest.java:296)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at ee.jakarta.tck.persistence.core.StoredProcedureQuery.Client1.main(Client1.java:67)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err] Caused by: java.lang.ClassNotFoundException: foo
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at org.glassfish.common.util.GlassfishUrlClassLoader.findClass(GlassfishUrlClassLoader.java:87)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at org.glassfish.appclient.client.acc.ACCClassLoader.findClassUnshadowed(ACCClassLoader.java:157)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at org.glassfish.appclient.client.acc.ACCClassLoader.findClass(ACCClassLoader.java:136)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:592)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at org.glassfish.common.util.GlassfishUrlClassLoader.loadClass(GlassfishUrlClassLoader.java:115)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at java.base/java.lang.Class.forName0(Native Method)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at java.base/java.lang.Class.forName(Class.java:467)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at java.naming/com.sun.naming.internal.VersionHelper.loadClass(VersionHelper.java:128)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at java.naming/com.sun.naming.internal.VersionHelper.loadClass(VersionHelper.java:134)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at java.naming/com.sun.naming.internal.VersionHelper.loadClass(VersionHelper.java:97)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   at java.naming/javax.naming.spi.NamingManager.getFactory(NamingManager.java:759)
> Oct 25, 2024 3:00:08 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err]   ... 14 more
> Oct 25, 2024 3:00:09 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err] Oct 25, 2024 3:00:09 PM org.glassfish.enterprise.iiop.impl.GlassFishORBManager getORB
> Oct 25, 2024 3:00:09 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
> INFO: [APPCLIENT-err] INFO: ORB initialization succeeded: [com.sun.corba.ee](http://com.sun.corba.ee/).impl.orb.ORBImpl@342ee097
> [3:04](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729883079444399?thread_ts=1729612177.243759&cid=C0131MLD538)
>  INFO: [APPCLIENT-err] Caused by: java.lang.ClassNotFoundException: foo
> 
> 
> Scott Marlow
>   Friday at 3:23 PM
> No calls make it into EJBObjectInvocationHandler
> 
> 
> Brian Decker
>   [Friday at 3:40 PM](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1729885259222019?thread_ts=1729612177.243759&cid=C0131MLD538)
> What's with the references to EJBHome you stepped through, I wonder? Is that just a misnomer on Glassfish's part? There shouldn't be anything like a Home interface in any of this stuff anymore.
> 
> 
> Scott Marlow
>   Friday at 5:05 PM
> I assume that was just legacy code to handle EJBHome that is still checked for.  I basically set a breakpoint on every exception thrown just to see more info which led to the ioexception which I'm guessing is caused by a protocol error but hard to tell

The current main must be missing some updates as I had to make a few changes just to get to the point of seeing the naming exception. I has pushed the changes to an appclient-jndi branch of https://github.com/jakartaee/platform-tck.git.

This is the current exception output:

```
10-31-2024 13:27:16:  ERROR: NamingException looking up vehicle: com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF
10-31-2024 13:27:16:  ERROR: javax.naming.NamingException: Lookup failed for java:comp/env/ejb/Stateless3VehicleBean in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: No object bound for java:comp/env/ejb/Stateless3VehicleBean]
        at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:836)
        at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:257)
        at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
        at com.sun.ts.lib.util.TSNamingContext.lookup(TSNamingContext.java:70)
        at com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleRunner.run(Stateless3VehicleRunner.java:45)
        at com.sun.ts.lib.harness.ServiceEETest.run(ServiceEETest.java:128)
        at com.sun.ts.lib.harness.EETest.getPropsReady(EETest.java:494)
        at com.sun.ts.lib.harness.ServiceEETest.run(ServiceEETest.java:232)
        at com.sun.ts.lib.harness.EETest.run(EETest.java:296)
        at ee.jakarta.tck.persistence.core.EntityGraph.Client.main(Client.java:48)
Caused by: javax.naming.NameNotFoundException: No object bound for java:comp/env/ejb/Stateless3VehicleBean
        at com.sun.enterprise.naming.impl.JavaURLContext.lookup(JavaURLContext.java:135)
        at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:818)
        ... 9 more
        Suppressed: javax.naming.NamingException: Exception resolving Ejb for 'Remote ejb-ref name=ejb/Stateless3VehicleBean,Remote 3.x interface =com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF,ejb-link=Stateless3VehicleBean,lookup=,mappedName=,jndi-name=,refType=Session' .  Actual (possibly internal) Remote JNDI name used for lookup is '#com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF' [Root exception is javax.naming.NamingException: Lookup failed for #com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: #com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF not found]]
                at com.sun.ejb.EjbNamingReferenceManagerImpl.resolveEjbReference(EjbNamingReferenceManagerImpl.java:159)
                at com.sun.enterprise.container.common.impl.ComponentEnvManagerImpl$EjbReferenceProxy.create(ComponentEnvManagerImpl.java:975)
                at com.sun.enterprise.naming.impl.GlassfishNamingManagerImpl.lookup(GlassfishNamingManagerImpl.java:621)
                at com.sun.enterprise.naming.impl.GlassfishNamingManagerImpl.lookup(GlassfishNamingManagerImpl.java:587)
                at com.sun.enterprise.naming.impl.JavaURLContext.lambda$lookup$0(JavaURLContext.java:153)
                at com.sun.enterprise.naming.impl.JavaURLContext.lookupOrCollectException(JavaURLContext.java:550)
                at com.sun.enterprise.naming.impl.JavaURLContext.lookup(JavaURLContext.java:153)
                ... 10 more
        Caused by: javax.naming.NamingException: Lookup failed for #com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: #com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF not found]
                at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:836)
                at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:257)
                at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
                at com.sun.ejb.EjbNamingReferenceManagerImpl.resolveEjbReference(EjbNamingReferenceManagerImpl.java:154)
                ... 16 more
        Caused by: javax.naming.NameNotFoundException: #com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF not found
                at com.sun.enterprise.naming.impl.TransientContext.doLookup(TransientContext.java:234)
                at com.sun.enterprise.naming.impl.TransientContext.lookup(TransientContext.java:202)
                at com.sun.enterprise.naming.impl.SerialContextProviderImpl.lookup(SerialContextProviderImpl.java:37)
                at com.sun.enterprise.naming.impl.RemoteSerialContextProviderImpl.lookup(RemoteSerialContextProviderImpl.java:85)
                at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
                at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                at java.base/java.lang.reflect.Method.invoke(Method.java:568)
                at com.sun.corba.ee.impl.presentation.rmi.ReflectiveTie.dispatchToMethod(ReflectiveTie.java:122)
                at com.sun.corba.ee.impl.presentation.rmi.ReflectiveTie._invoke(ReflectiveTie.java:152)
                at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatchToServant(ServerRequestDispatcherImpl.java:501)
                at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatch(ServerRequestDispatcherImpl.java:172)
                at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequestRequest(MessageMediatorImpl.java:1529)
                at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequest(MessageMediatorImpl.java:1405)
                at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleInput(MessageMediatorImpl.java:910)
                at com.sun.corba.ee.impl.protocol.giopmsgheaders.RequestMessage_1_2.callback(RequestMessage_1_2.java:192)
                at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequest(MessageMediatorImpl.java:674)
                at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.dispatch(MessageMediatorImpl.java:476)
                at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.doWork(MessageMediatorImpl.java:2202)
                at com.sun.corba.ee.impl.threadpool.ThreadPoolImpl$WorkerThread.performWork(ThreadPoolImpl.java:476)
                at com.sun.corba.ee.impl.threadpool.ThreadPoolImpl$WorkerThread.run(ThreadPoolImpl.java:519)

[2024-10-31 13:27:16] F org.glassfish.naming getInitialContext(env={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming:com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl:com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}) 
10-31-2024 13:27:16:  ERROR: TSNamingContext.CS: (file:/Users/starksm/Dev/Jakarta/platform-tck/glassfish-runner/jpa-platform-tck/target/lib/arquillian-protocol-lib.jar <no signer certificates>)

1. AppManagedVehicleRunner jndi dump walking down tree branch name = 
2. AppManagedVehicleRunner jndi dump (show name  classname pair): com_sun_ts_tests_common_vehicle_Stateless3VehicleBean#com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF: javax.naming.Reference
[2024-10-31 13:27:16] F org.glassfish.naming getObjectInstance(name=com_sun_ts_tests_common_vehicle_Stateless3VehicleBean#com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF, obj=Reference Class Name: com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF
Type: url
Content: com_sun_ts_tests_common_vehicle_Stateless3VehicleBean__3_x_Internal_RemoteBusinessHome__
) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=com.sun.ejb.containers.RemoteBusinessObjectFactory, resolve=false) 
[2024-10-31 13:27:16] F org.glassfish.naming getInitialContext(env={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming:com.sun.enterprise.naming:com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl:com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl:com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=com.sun.ejb.codegen.GenericEJBHome_Generated, resolve=false) 
[2024-10-31 13:27:16] F org.glassfish.naming getObjectInstance(name=com_sun_ts_tests_common_vehicle_Stateless3VehicleBean__3_x_Internal_RemoteBusinessHome__, obj=IORImpl[type=RMI:com.sun.ejb.codegen.GenericEJBHome_Generated:0000000000000000 iorTemplates=[IORTemplate[oktemp=ObjectKeyTemplate[magic=afabcb00 scid=38 serverid=100 orbid=S1AS-ORB oaid=ObjectAdapterID[RootPOA/#RFMBase#/113403680031309824_RBusiness_com.sun.ts.tests.common.vehicle.stateless3._Stateless3VehicleIF_Remote-EJBHome] profile templates:[IIOPProfileTemplateImpl[giopVersion=1.2 primary=127.0.0.1:3700]]]]) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=com.sun.ejb.codegen.GenericEJBHome_Generated, resolve=false) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=com.sun.ejb.codegen.GenericEJBHome_Generated, resolve=false) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=com.sun.ejb.containers.GenericEJBHome, resolve=false) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=java.lang.Object, resolve=false) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=java.lang.String, resolve=false) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=java.rmi.Remote, resolve=false) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=java.rmi.RemoteException, resolve=false) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=javax.rmi.CORBA.Stub, resolve=false) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=org.omg.CORBA_2_3.portable.ObjectImpl, resolve=false) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=org.omg.CORBA.portable.ObjectImpl, resolve=false) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=java.lang.Long, resolve=false) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=com.sun.corba.ee.impl.presentation.rmi.codegen.CodegenStubBase, resolve=false) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF, resolve=false) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=com.sun.ts.tests.common.vehicle.stateless3._Stateless3VehicleIF_Wrapper, resolve=false) 
2. AppManagedVehicleRunner jndi dump (show name  classname pair): com_sun_ts_tests_common_vehicle_Stateless3VehicleBean: javax.naming.Reference
[2024-10-31 13:27:16] F org.glassfish.naming getObjectInstance(name=com_sun_ts_tests_common_vehicle_Stateless3VehicleBean, obj=Reference Class Name: com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF
Type: url
Content: com_sun_ts_tests_common_vehicle_Stateless3VehicleBean__3_x_Internal_RemoteBusinessHome__
) 
[2024-10-31 13:27:16] F org.glassfish.naming getInitialContext(env={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming:com.sun.enterprise.naming:com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl:com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl:com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}) 
[2024-10-31 13:27:16] F org.glassfish.naming getObjectInstance(name=com_sun_ts_tests_common_vehicle_Stateless3VehicleBean__3_x_Internal_RemoteBusinessHome__, obj=IORImpl[type=RMI:com.sun.ejb.codegen.GenericEJBHome_Generated:0000000000000000 iorTemplates=[IORTemplate[oktemp=ObjectKeyTemplate[magic=afabcb00 scid=38 serverid=100 orbid=S1AS-ORB oaid=ObjectAdapterID[RootPOA/#RFMBase#/113403680031309824_RBusiness_com.sun.ts.tests.common.vehicle.stateless3._Stateless3VehicleIF_Remote-EJBHome] profile templates:[IIOPProfileTemplateImpl[giopVersion=1.2 primary=127.0.0.1:3700]]]]) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=com.sun.ejb.codegen.GenericEJBHome_Generated, resolve=false) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF, resolve=false) 
[2024-10-31 13:27:16] F org.glassfish.common.util.GlassfishUrlClassLoader loadClass(name=com.sun.ts.tests.common.vehicle.stateless3._Stateless3VehicleIF_Wrapper, resolve=false) 
2. AppManagedVehicleRunner jndi dump (show name  classname pair): com_sun_ts_tests_common_vehicle_Stateless3VehicleBean__3_x_Internal_RemoteBusinessHome__: javax.naming.Reference
[2024-10-31 13:27:16] F org.glassfish.naming getObjectInstance(name=com_sun_ts_tests_common_vehicle_Stateless3VehicleBean__3_x_Internal_RemoteBusinessHome__, obj=IORImpl[type=RMI:com.sun.ejb.codegen.GenericEJBHome_Generated:0000000000000000 iorTemplates=[IORTemplate[oktemp=ObjectKeyTemplate[magic=afabcb00 scid=38 serverid=100 orbid=S1AS-ORB oaid=ObjectAdapterID[RootPOA/#RFMBase#/113403680031309824_RBusiness_com.sun.ts.tests.common.vehicle.stateless3._Stateless3VehicleIF_Remote-EJBHome] profile templates:[IIOPProfileTemplateImpl[giopVersion=1.2 primary=127.0.0.1:3700]]]]) 
10-31-2024 13:27:16:  ERROR: Test failed.
```

It looks like the appclient env is being linked to the Stateless3VehicleBean, but the resolution of the bound object factory for the ejb-ref somehow ends up trying to lookup "#com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF", which is not a correct server side binding. 

The application-client.xml in the test artifact is:
```
<application-client xmlns="https://jakarta.ee/xml/ns/jakartaee" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="10" xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee https://jakarta.ee/xml/ns/jakartaee/application-client_10.xsd">
  <ejb-ref>
    <ejb-ref-name>ejb/Stateless3VehicleBean</ejb-ref-name>
    <ejb-ref-type>Session</ejb-ref-type>
    <remote>com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF</remote>
    <ejb-link>Stateless3VehicleBean</ejb-link>
  </ejb-ref>
</application-client>
```

and the sun-ejb-jar.xml file (which the glassfish docs says it supports) is:
```
<sun-ejb-jar>
    <enterprise-beans>
    <ejb>
      <ejb-name>Stateless3VehicleBean</ejb-name>
      <jndi-name>com_sun_ts_tests_common_vehicle_Stateless3VehicleBean</jndi-name>
    </ejb>
    </enterprise-beans>
</sun-ejb-jar>
```

If I can the ejb-name in this file, the deployment fails with an exception of:
```
Caused by: org.omnifaces.arquillian.jersey.server.ContainerException: While Deploying Application: jpa_core_EntityGraph_vehicles --exit_code: FAILURE, message: Error occurred during deployment: Exception while deploying the app [jpa_core_EntityGraph_vehicles] : Referencing error: this bundle has no bean of name: ejb/Stateless3VehicleBean. Please see server.log for more details. [status: CLIENT_ERROR reason: Bad Request]
```

so the current test artifact deployment that succeeds is finding a match for the appclient ejb-link. The bound object factory is just failing to resolve the remote lookup at runtime for some reason.

I'll double check the EE 10 deployment artifact to see if there is some other descriptor missing.

@arjantijms @dmatej 

I have created a simple reproducer that shows the problem. The appclient has an entry in the java:comp/env/ejb/StatelessVehicleBean location, but the EjbReferenceProxy value bound there ends up attempting to lookup an invalid jndi name. To run the reproducer:

Clone the https://github.com/jakartaee/platform-tck.git
cd platform-tck
git checkout appclient-jndi
cd glassfish-runner/jpa-platform-tck
mvn -Darquillian.xml=test-arquillian.xml test

This produces a verbose appclient container log at glassfish-runner/jpa-platform-tck/target/appclient-0.log

```
Exception in thread "main" javax.naming.NamingException: Lookup failed for java:comp/env/ejb/StatelessVehicleBean in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: No object bound for java:comp/env/ejb/StatelessVehicleBean]
        at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:836)
        at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:257)
        at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
        at client.RemoteEjbClient.main(RemoteEjbClient.java:10)
Caused by: javax.naming.NameNotFoundException: No object bound for java:comp/env/ejb/StatelessVehicleBean
        at com.sun.enterprise.naming.impl.JavaURLContext.lookup(JavaURLContext.java:135)
        at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:818)
        ... 3 more
        Suppressed: javax.naming.NamingException: Exception resolving Ejb for 'Remote ejb-ref name=ejb/StatelessVehicleBean,Remote 3.x interface =ejb.VehicleIF,ejb-link=StatelessVehicleBean,lookup=,mappedName=,jndi-name=,refType=Session' .  Actual (possibly internal) Remote JNDI name used for lookup is '#ejb.VehicleIF' [Root exception is javax.naming.NamingException: Lookup failed for #ejb.VehicleIF in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: #ejb.VehicleIF not found]]
                at com.sun.ejb.EjbNamingReferenceManagerImpl.resolveEjbReference(EjbNamingReferenceManagerImpl.java:159)
                at com.sun.enterprise.container.common.impl.ComponentEnvManagerImpl$EjbReferenceProxy.create(ComponentEnvManagerImpl.java:975)
                at com.sun.enterprise.naming.impl.GlassfishNamingManagerImpl.lookup(GlassfishNamingManagerImpl.java:621)
                at com.sun.enterprise.naming.impl.GlassfishNamingManagerImpl.lookup(GlassfishNamingManagerImpl.java:587)
                at com.sun.enterprise.naming.impl.JavaURLContext.lambda$lookup$0(JavaURLContext.java:153)
                at com.sun.enterprise.naming.impl.JavaURLContext.lookupOrCollectException(JavaURLContext.java:550)
                at com.sun.enterprise.naming.impl.JavaURLContext.lookup(JavaURLContext.java:153)
                ... 4 more
        Caused by: javax.naming.NamingException: Lookup failed for #ejb.VehicleIF in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: #ejb.VehicleIF not found]
                at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:836)
                at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:257)
                at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
                at com.sun.ejb.EjbNamingReferenceManagerImpl.resolveEjbReference(EjbNamingReferenceManagerImpl.java:154)
                ... 10 more
        Caused by: javax.naming.NameNotFoundException: #ejb.VehicleIF not found
                at com.sun.enterprise.naming.impl.TransientContext.doLookup(TransientContext.java:234)
                at com.sun.enterprise.naming.impl.TransientContext.lookup(TransientContext.java:202)
                at com.sun.enterprise.naming.impl.SerialContextProviderImpl.lookup(SerialContextProviderImpl.java:37)
                at com.sun.enterprise.naming.impl.RemoteSerialContextProviderImpl.lookup(RemoteSerialContextProviderImpl.java:85)
                at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
                at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                at java.base/java.lang.reflect.Method.invoke(Method.java:568)
                at com.sun.corba.ee.impl.presentation.rmi.ReflectiveTie.dispatchToMethod(ReflectiveTie.java:122)
                at com.sun.corba.ee.impl.presentation.rmi.ReflectiveTie._invoke(ReflectiveTie.java:152)
                at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatchToServant(ServerRequestDispatcherImpl.java:501)
                at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatch(ServerRequestDispatcherImpl.java:172)
                at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequestRequest(MessageMediatorImpl.java:1529)
                at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequest(MessageMediatorImpl.java:1405)
                at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleInput(MessageMediatorImpl.java:910)
                at com.sun.corba.ee.impl.protocol.giopmsgheaders.RequestMessage_1_2.callback(RequestMessage_1_2.java:192)
                at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequest(MessageMediatorImpl.java:674)
                at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.dispatch(MessageMediatorImpl.java:476)
                at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.doWork(MessageMediatorImpl.java:2202)
                at com.sun.corba.ee.impl.threadpool.ThreadPoolImpl$WorkerThread.performWork(ThreadPoolImpl.java:476)
                at com.sun.corba.ee.impl.threadpool.ThreadPoolImpl$WorkerThread.run(ThreadPoolImpl.java:519)
```

> git checkout appclient-jndi

I don't see such branch, but I can see your branch `correct_jpa_sunappclientxml`, is it the right one?

Expectation: having built glassfish 8.0.0-SNAPSHOT and having the clone of the jakartaee-tck repo.
What I did:
```
git checkout correct_jpa_sunappclientxml 
mvn clean install
mvn clean install -f glassfish-runner/jpa-platform-tck -Darquillian.xml=test-arquillian.xml -Dglassfish.container.version=8.0.0-SNAPSHOT
```
... and the same result with the default glassfish version 8.0.0-JDK17-M7
``` 
....
[INFO] --- antrun:3.0.0:run (005-initdb) @ glassfish.jpa-platform-tck ---
[INFO] Executing tasks
[INFO]       [sql] Executing resource: /home/dmatej/work/repo/git/jakartaee-tck/glassfish-runner/jpa-platform-tck/sql/derby/derby.ddl.sql
[ERROR]       [sql] Failed to execute:  ALTER TABLE PHONE_TABLE DROP CONSTRAINT FK_FOR_ADDRESS 
[ERROR]       [sql] java.sql.SQLSyntaxErrorException: Schema 'CTS1' does not exist
....
[INFO] Running ee.jakarta.tck.persistence.core.EntityGraph.ClientAppmanagedTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.069 s <<< FAILURE! -- in ee.jakarta.tck.persistence.core.EntityGraph.ClientAppmanagedTest
[ERROR] ee.jakarta.tck.persistence.core.EntityGraph.ClientAppmanagedTest -- Time elapsed: 0.069 s <<< ERROR!
java.lang.RuntimeException: Could not create new instance of class org.jboss.arquillian.test.impl.EventTestRunnerAdaptor
        at org.jboss.arquillian.test.spi.SecurityActions.newInstance(SecurityActions.java:146)
        at org.jboss.arquillian.test.spi.SecurityActions.newInstance(SecurityActions.java:89)
        at org.jboss.arquillian.test.spi.TestRunnerAdaptorBuilder.build(TestRunnerAdaptorBuilder.java:49)
        at org.jboss.arquillian.junit5.JUnitJupiterTestClassLifecycleManager.initializeAdaptor(JUnitJupiterTestClassLifecycleManager.java:38)
        at org.jboss.arquillian.junit5.JUnitJupiterTestClassLifecycleManager.getManager(JUnitJupiterTestClassLifecycleManager.java:25)
        at org.jboss.arquillian.junit5.ArquillianExtension.beforeAll(ArquillianExtension.java:35)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
        Suppressed: java.lang.RuntimeException: Arquillian initialization has already been attempted, but failed. See previous exceptions for cause
                at org.jboss.arquillian.junit5.JUnitJupiterTestClassLifecycleManager.handleSuiteLevelFailure(JUnitJupiterTestClassLifecycleManager.java:62)
                at org.jboss.arquillian.junit5.JUnitJupiterTestClassLifecycleManager.getManager(JUnitJupiterTestClassLifecycleManager.java:30)
                at org.jboss.arquillian.junit5.ArquillianExtension.afterAll(ArquillianExtension.java:42)
                ... 1 more
        Caused by: [CIRCULAR REFERENCE: java.lang.RuntimeException: Could not create new instance of class org.jboss.arquillian.test.impl.EventTestRunnerAdaptor]
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
        at org.jboss.arquillian.test.spi.SecurityActions.newInstance(SecurityActions.java:144)
        ... 6 more
Caused by: java.lang.IllegalArgumentException: Could not find configured filename as either classpath resource nor file resource: test-arquillian.xml
        at org.jboss.arquillian.config.impl.extension.FileUtils.loadResource(FileUtils.java:63)
        at org.jboss.arquillian.config.impl.extension.FileUtils.loadArquillianXml(FileUtils.java:51)
        at org.jboss.arquillian.config.impl.extension.ConfigurationRegistrar.loadConfiguration(ConfigurationRegistrar.java:75)
        at org.jboss.arquillian.config.impl.extension.ConfigurationRegistrar.loadConfiguration(ConfigurationRegistrar.java:62)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
        at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
        at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
        at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:134)
        at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:106)
        at org.jboss.arquillian.core.impl.ManagerImpl.start(ManagerImpl.java:254)
        at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.<init>(EventTestRunnerAdaptor.java:61)
        ... 9 more

```

The appclient-jndi branch is in the main platform-tck repo. I just tried a clean checkout and test run:

```
starksm@Scotts-Mac-Studio /tmp % rm -rf platform-tck 
starksm@Scotts-Mac-Studio /tmp % git clone https://github.com/jakartaee/platform-tck.git
Cloning into 'platform-tck'...
remote: Enumerating objects: 190159, done.
remote: Counting objects: 100% (20725/20725), done.
remote: Compressing objects: 100% (3951/3951), done.
remote: Total 190159 (delta 15677), reused 19916 (delta 15194), pack-reused 169434 (from 1)
Receiving objects: 100% (190159/190159), 103.20 MiB | 2.53 MiB/s, done.
Resolving deltas: 100% (139978/139978), done.
Updating files: 100% (22544/22544), done.
starksm@Scotts-Mac-Studio /tmp % cd platform-tck 
starksm@Scotts-Mac-Studio platform-tck % git checkout appclient-jndi
Branch 'appclient-jndi' set up to track remote branch 'appclient-jndi' from 'origin'.
Switched to a new branch 'appclient-jndi'
starksm@Scotts-Mac-Studio platform-tck %  mvn -Dmaven.build.cache.enabled=false test -f glassfish-runner/jpa-platform-tck -Darquillian.xml=test-arquillian.xml -Dglassfish.container.version=8.0.0-SNAPSHOT
[INFO] Cache disabled by command line flag, project will be built fully and not cached
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for jakarta:glassfish.jpa-platform-tck:jar:11.0.0-SNAPSHOT
[WARNING] 'build.plugins.plugin.(groupId:artifactId)' must be unique but found duplicate declaration of plugin org.apache.maven.plugins:maven-dependency-plugin @ line 366, column 21
[WARNING] 'build.plugins.plugin.(groupId:artifactId)' must be unique but found duplicate declaration of plugin org.apache.maven.plugins:maven-antrun-plugin @ line 492, column 21
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] 
[INFO] -----------------< jakarta:glassfish.jpa-platform-tck >-----------------
[INFO] Building glassfish.jpa-platform-tck 11.0.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- enforcer:3.0.0:enforce (enforce-maven) @ glassfish.jpa-platform-tck ---
[INFO] 
[INFO] --- spotless:2.36.0:check (check-spotless-poms) @ glassfish.jpa-platform-tck ---
[INFO] Index file does not exist. Fallback to an empty index
[INFO] Sorting file /var/folders/yh/t0fnl559607crz2vyp509bjr0000gn/T/pom16316726856461708360.xml
[INFO] Pom file is already sorted, exiting
[INFO] Sorting file /var/folders/yh/t0fnl559607crz2vyp509bjr0000gn/T/pom1542785978384178646.xml
[INFO] Pom file is already sorted, exiting
[INFO] Spotless.Pom is keeping 1 files clean - 0 needs changes to be clean, 1 were already clean, 0 were skipped because caching determined they were already clean
[INFO] 
[INFO] --- dependency:3.7.1:unpack (001-unpack) @ glassfish.jpa-platform-tck ---
[INFO] Configured Artifact: org.glassfish.main.distributions:glassfish:8.0.0-SNAPSHOT:zip
[INFO] 
[INFO] --- dependency:3.7.1:copy (003-copy-protocol-lib) @ glassfish.jpa-platform-tck ---
[INFO] Configured Artifact: jakarta.tck.arquillian:arquillian-protocol-lib:?:jar
[INFO] Copying artifact 'jakarta.tck.arquillian:arquillian-protocol-lib:jar:11.0.0-SNAPSHOT' (/Users/starksm/.m2/repository/jakarta/tck/arquillian/arquillian-protocol-lib/11.0.0-SNAPSHOT/arquillian-protocol-lib-11.0.0-SNAPSHOT.jar) to /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/protocol/protocol.jar
[INFO] 
[INFO] --- dependency:3.7.1:copy (004-copy-lib) @ glassfish.jpa-platform-tck ---
[INFO] Configured Artifact: jakarta.tck:libutil:?:jar
[INFO] Configured Artifact: jakarta.tck:common:?:jar
[INFO] Configured Artifact: jakarta.tck.arquillian:tck-porting-lib:?:jar
[INFO] Configured Artifact: jakarta.tck:runtime:?:jar
[INFO] Configured Artifact: jakarta.tck:persistence-platform-tck-tests:?:jar
[INFO] Configured Artifact: org.apache.derby:derbyclient:?:jar
[INFO] Configured Artifact: org.apache.derby:derbytools:?:jar
[INFO] Configured Artifact: jakarta.tck.arquillian:arquillian-protocol-lib:?:jar
[INFO] Copying artifact 'jakarta.tck:libutil:jar:11.0.0-SNAPSHOT' (/Users/starksm/.m2/repository/jakarta/tck/libutil/11.0.0-SNAPSHOT/libutil-11.0.0-SNAPSHOT.jar) to /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/lib/libutil.jar
[INFO] Copying artifact 'jakarta.tck:common:jar:11.0.0-SNAPSHOT' (/Users/starksm/.m2/repository/jakarta/tck/common/11.0.0-SNAPSHOT/common-11.0.0-SNAPSHOT.jar) to /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/lib/common.jar
[INFO] Copying artifact 'jakarta.tck.arquillian:tck-porting-lib:jar:11.0.0-SNAPSHOT' (/Users/starksm/.m2/repository/jakarta/tck/arquillian/tck-porting-lib/11.0.0-SNAPSHOT/tck-porting-lib-11.0.0-SNAPSHOT.jar) to /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/lib/tck-porting-lib.jar
[INFO] Copying artifact 'jakarta.tck:runtime:jar:11.0.0-SNAPSHOT' (/Users/starksm/.m2/repository/jakarta/tck/runtime/11.0.0-SNAPSHOT/runtime-11.0.0-SNAPSHOT.jar) to /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/lib/runtime.jar
[INFO] Copying artifact 'jakarta.tck:persistence-platform-tck-tests:jar:11.0.0-SNAPSHOT' (/Users/starksm/.m2/repository/jakarta/tck/persistence-platform-tck-tests/11.0.0-SNAPSHOT/persistence-platform-tck-tests-11.0.0-SNAPSHOT.jar) to /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/lib/persistence-platform-tck-tests.jar
[INFO] Copying artifact 'org.apache.derby:derbyclient:jar:10.15.2.0' (/Users/starksm/.m2/repository/org/apache/derby/derbyclient/10.15.2.0/derbyclient-10.15.2.0.jar) to /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/lib/derbyclient.jar
[INFO] Copying artifact 'org.apache.derby:derbytools:jar:10.15.2.0' (/Users/starksm/.m2/repository/org/apache/derby/derbytools/10.15.2.0/derbytools-10.15.2.0.jar) to /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/lib/derbytools.jar
[INFO] Copying artifact 'jakarta.tck.arquillian:arquillian-protocol-lib:jar:11.0.0-SNAPSHOT' (/Users/starksm/.m2/repository/jakarta/tck/arquillian/arquillian-protocol-lib/11.0.0-SNAPSHOT/arquillian-protocol-lib-11.0.0-SNAPSHOT.jar) to /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/lib/arquillian-protocol-lib.jar
[INFO] 
[INFO] --- antrun:3.0.0:run (002-asadmin-permission) @ glassfish.jpa-platform-tck ---
[INFO] Executing tasks
[INFO] Executed tasks
[INFO] 
[INFO] --- resources:3.3.1:resources (default-resources) @ glassfish.jpa-platform-tck ---
[INFO] Copying 1 resource from src/main/resources to target/classes
[INFO] 
[INFO] --- exec:3.2.0:exec (004-StartDatabase) @ glassfish.jpa-platform-tck ---
Database already started on host 0.0.0.0, port 1527.
Command start-database executed successfully.
[INFO] 
[INFO] --- compiler:3.12.1:compile (default-compile) @ glassfish.jpa-platform-tck ---
[INFO] Recompiling the module because of changed source code.
[INFO] Compiling 2 source files with javac [debug release 17] to target/classes
[INFO] /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/src/main/java/org/glassfish/transactions/core/tck/GlassfishTestArchiveProcessor.java: /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/src/main/java/org/glassfish/transactions/core/tck/GlassfishTestArchiveProcessor.java uses or overrides a deprecated API.
[INFO] /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/src/main/java/org/glassfish/transactions/core/tck/GlassfishTestArchiveProcessor.java: Recompile with -Xlint:deprecation for details.
[INFO] 
[INFO] --- resources:3.3.1:testResources (default-testResources) @ glassfish.jpa-platform-tck ---
[INFO] Copying 5 resources from src/test/resources to target/test-classes
[INFO] 
[INFO] --- compiler:3.12.1:testCompile (default-testCompile) @ glassfish.jpa-platform-tck ---
[INFO] Recompiling the module because of changed dependency.
[INFO] Compiling 4 source files with javac [debug release 17] to target/test-classes
[INFO] 
[INFO] --- surefire:3.0.0:test (default-test) @ glassfish.jpa-platform-tck ---
[INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running client.AppClientTest
Created appclient directory: /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/appclient
Exported test ear to: /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/appclient/jpa_core_EntityGraph_vehicles.ear
Exported test ear content to: /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/appclient/jpa_core_EntityGraph_stateless_vehicle_ejb.jar
Exported test ear content to: /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/appclient/jpa_core_EntityGraph_vehicles_client.jar
Starting container using command: [/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home/bin/java, -jar, /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/modules/admin-cli.jar, start-domain, -t]
Waiting finished after 1,740 ms.
Successfully started the domain : domain1
domain  Location: /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/domains/domain1
Log File: /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/domains/domain1/logs/server.log
Admin Port: 4,848
Nov 06, 2024 12:48:06 PM tck.arquillian.protocol.appclient.AppClientCmd run
INFO: Created process[user: Optional[starksm], cmd: /bin/sh, args: [target/glassfish8/glassfish/bin/appclient, -Djava.util.logging.config.file=jakartaeetck/logging.properties, -jar, /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/appclient/jpa_core_EntityGraph_vehicles_client.jar], startTime: Optional[2024-11-06T18:48:06.134Z]]
Nov 06, 2024 12:48:06 PM tck.arquillian.protocol.appclient.AppClientCmd run
INFO: process(11659).envp: [AS_JAVA=/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home, AS_DEBUG=true]
Nov 06, 2024 12:48:06 PM tck.arquillian.protocol.appclient.AppClientCmd readClientProcess
INFO: Begin readClientProcess
Nov 06, 2024 12:48:06 PM tck.arquillian.protocol.appclient.AppClientCmd readClientProcess
INFO: Begin readClientProcess
Nov 06, 2024 12:48:06 PM tck.arquillian.protocol.appclient.AppClientCmd run
INFO: Started process reader threads
Nov 06, 2024 12:48:06 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] acc._AS_INSTALL set to target/glassfish8/glassfish/bin/..
Nov 06, 2024 12:48:06 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] VMARGS = null
Nov 06, 2024 12:48:06 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] "/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home/bin/java" -Dorg.glassfish.gmbal.no.multipleUpperBoundsException=true --add-opens=java.base/java.lang=ALL-UNNAMED -Dcom.sun.aas.installRoot="/private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/.." -Djava.security.policy="/private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/../lib/appclient/client.policy" -classpath /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/../lib/gf-client.jar:. -Djava.system.class.loader=org.glassfish.appclient.client.acc.agent.ACCAgentClassLoader -Xshare:off -Djava.security.auth.login.config="/private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/../lib/appclient/appclientlogin.conf"  "-Djava.util.logging.config.file=jakartaeetck/logging.properties"  -javaagent:"/private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/../lib/gf-client.jar"=argsfile="/var/folders/yh/t0fnl559607crz2vyp509bjr0000gn/T/acc484379618692860999.dat"  -classpath  /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/../lib/gf-client.jar:/private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/appclient/jpa_core_EntityGraph_vehicles_client.jar  client.RemoteEjbClient 
Nov 06, 2024 12:48:06 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] [2024-11-06 12:48:06] I jakarta.enterprise.resource.jta.com.sun.enterprise.transaction DTX5019: Transaction Manager is ready. Using [com.sun.enterprise.transaction.jts.JavaEETransactionManagerJTSDelegate] as the delegate 
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] [2024-11-06 12:48:07] I org.glassfish.enterprise.iiop.impl.GlassFishORBManager ORB initialization succeeded: com.sun.corba.ee.impl.orb.ORBImpl@4f3356c0 
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] Exception in thread "main" javax.naming.NamingException: Lookup failed for java:comp/env/ejb/StatelessVehicleBean in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: No object bound for java:comp/env/ejb/StatelessVehicleBean]
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 	at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:836)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 	at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:257)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 	at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 	at client.RemoteEjbClient.main(RemoteEjbClient.java:13)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] Caused by: javax.naming.NameNotFoundException: No object bound for java:comp/env/ejb/StatelessVehicleBean
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 	at com.sun.enterprise.naming.impl.JavaURLContext.lookup(JavaURLContext.java:135)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 	at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:818)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 	... 3 more
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 	Suppressed: javax.naming.NamingException: Exception resolving Ejb for 'Remote ejb-ref name=ejb/StatelessVehicleBean,Remote 3.x interface =ejb.VehicleIF,ejb-link=StatelessVehicleBean,lookup=,mappedName=,jndi-name=,refType=Session' .  Actual (possibly internal) Remote JNDI name used for lookup is '#ejb.VehicleIF' [Root exception is javax.naming.NamingException: Lookup failed for #ejb.VehicleIF in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: #ejb.VehicleIF not found]]
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.ejb.EjbNamingReferenceManagerImpl.resolveEjbReference(EjbNamingReferenceManagerImpl.java:159)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.enterprise.container.common.impl.ComponentEnvManagerImpl$EjbReferenceProxy.create(ComponentEnvManagerImpl.java:975)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.enterprise.naming.impl.GlassfishNamingManagerImpl.lookup(GlassfishNamingManagerImpl.java:621)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.enterprise.naming.impl.GlassfishNamingManagerImpl.lookup(GlassfishNamingManagerImpl.java:587)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.enterprise.naming.impl.JavaURLContext.lambda$lookup$0(JavaURLContext.java:153)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.enterprise.naming.impl.JavaURLContext.lookupOrCollectException(JavaURLContext.java:550)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.enterprise.naming.impl.JavaURLContext.lookup(JavaURLContext.java:153)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		... 4 more
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 	Caused by: javax.naming.NamingException: Lookup failed for #ejb.VehicleIF in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: #ejb.VehicleIF not found]
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:836)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:257)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.ejb.EjbNamingReferenceManagerImpl.resolveEjbReference(EjbNamingReferenceManagerImpl.java:154)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		... 10 more
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 	Caused by: javax.naming.NameNotFoundException: #ejb.VehicleIF not found
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.enterprise.naming.impl.TransientContext.doLookup(TransientContext.java:235)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.enterprise.naming.impl.TransientContext.lookup(TransientContext.java:203)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.enterprise.naming.impl.SerialContextProviderImpl.lookup(SerialContextProviderImpl.java:38)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.enterprise.naming.impl.RemoteSerialContextProviderImpl.lookup(RemoteSerialContextProviderImpl.java:92)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at java.base/java.lang.reflect.Method.invoke(Method.java:568)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.corba.ee.impl.presentation.rmi.ReflectiveTie.dispatchToMethod(ReflectiveTie.java:122)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.corba.ee.impl.presentation.rmi.ReflectiveTie._invoke(ReflectiveTie.java:152)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatchToServant(ServerRequestDispatcherImpl.java:501)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatch(ServerRequestDispatcherImpl.java:172)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequestRequest(MessageMediatorImpl.java:1529)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequest(MessageMediatorImpl.java:1405)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleInput(MessageMediatorImpl.java:910)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.corba.ee.impl.protocol.giopmsgheaders.RequestMessage_1_2.callback(RequestMessage_1_2.java:192)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequest(MessageMediatorImpl.java:674)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.dispatch(MessageMediatorImpl.java:476)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.doWork(MessageMediatorImpl.java:2202)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.corba.ee.impl.threadpool.ThreadPoolImpl$WorkerThread.performWork(ThreadPoolImpl.java:476)
Nov 06, 2024 12:48:07 PM tck.arquillian.protocol.appclient.AppClientCmd errorLineReceived
INFO: [APPCLIENT-err] 		at com.sun.corba.ee.impl.threadpool.ThreadPoolImpl$WorkerThread.run(ThreadPoolImpl.java:519)
Nov 06, 2024 12:48:08 PM tck.arquillian.protocol.appclient.AppClientCmd readClientProcess
INFO: Exiting(false), read 0 lines
Nov 06, 2024 12:48:08 PM tck.arquillian.protocol.appclient.AppClientCmd readClientProcess
INFO: Exiting(true), read 51 lines
acc._AS_INSTALL set to target/glassfish8/glassfish/bin/..
VMARGS = null
"/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home/bin/java" -Dorg.glassfish.gmbal.no.multipleUpperBoundsException=true --add-opens=java.base/java.lang=ALL-UNNAMED -Dcom.sun.aas.installRoot="/private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/.." -Djava.security.policy="/private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/../lib/appclient/client.policy" -classpath /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/../lib/gf-client.jar:. -Djava.system.class.loader=org.glassfish.appclient.client.acc.agent.ACCAgentClassLoader -Xshare:off -Djava.security.auth.login.config="/private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/../lib/appclient/appclientlogin.conf"  "-Djava.util.logging.config.file=jakartaeetck/logging.properties"  -javaagent:"/private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/../lib/gf-client.jar"=argsfile="/var/folders/yh/t0fnl559607crz2vyp509bjr0000gn/T/acc484379618692860999.dat"  -classpath  /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/bin/../lib/gf-client.jar:/private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/appclient/jpa_core_EntityGraph_vehicles_client.jar  client.RemoteEjbClient 
[2024-11-06 12:48:06] I jakarta.enterprise.resource.jta.com.sun.enterprise.transaction DTX5019: Transaction Manager is ready. Using [com.sun.enterprise.transaction.jts.JavaEETransactionManagerJTSDelegate] as the delegate 
[2024-11-06 12:48:07] I org.glassfish.enterprise.iiop.impl.GlassFishORBManager ORB initialization succeeded: com.sun.corba.ee.impl.orb.ORBImpl@4f3356c0 
Exception in thread "main" javax.naming.NamingException: Lookup failed for java:comp/env/ejb/StatelessVehicleBean in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: No object bound for java:comp/env/ejb/StatelessVehicleBean]
	at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:836)
	at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:257)
	at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
	at client.RemoteEjbClient.main(RemoteEjbClient.java:13)
Caused by: javax.naming.NameNotFoundException: No object bound for java:comp/env/ejb/StatelessVehicleBean
	at com.sun.enterprise.naming.impl.JavaURLContext.lookup(JavaURLContext.java:135)
	at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:818)
	... 3 more
	Suppressed: javax.naming.NamingException: Exception resolving Ejb for 'Remote ejb-ref name=ejb/StatelessVehicleBean,Remote 3.x interface =ejb.VehicleIF,ejb-link=StatelessVehicleBean,lookup=,mappedName=,jndi-name=,refType=Session' .  Actual (possibly internal) Remote JNDI name used for lookup is '#ejb.VehicleIF' [Root exception is javax.naming.NamingException: Lookup failed for #ejb.VehicleIF in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: #ejb.VehicleIF not found]]
		at com.sun.ejb.EjbNamingReferenceManagerImpl.resolveEjbReference(EjbNamingReferenceManagerImpl.java:159)
		at com.sun.enterprise.container.common.impl.ComponentEnvManagerImpl$EjbReferenceProxy.create(ComponentEnvManagerImpl.java:975)
		at com.sun.enterprise.naming.impl.GlassfishNamingManagerImpl.lookup(GlassfishNamingManagerImpl.java:621)
		at com.sun.enterprise.naming.impl.GlassfishNamingManagerImpl.lookup(GlassfishNamingManagerImpl.java:587)
		at com.sun.enterprise.naming.impl.JavaURLContext.lambda$lookup$0(JavaURLContext.java:153)
		at com.sun.enterprise.naming.impl.JavaURLContext.lookupOrCollectException(JavaURLContext.java:550)
		at com.sun.enterprise.naming.impl.JavaURLContext.lookup(JavaURLContext.java:153)
		... 4 more
	Caused by: javax.naming.NamingException: Lookup failed for #ejb.VehicleIF in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: #ejb.VehicleIF not found]
		at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:836)
		at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:257)
		at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
		at com.sun.ejb.EjbNamingReferenceManagerImpl.resolveEjbReference(EjbNamingReferenceManagerImpl.java:154)
		... 10 more
	Caused by: javax.naming.NameNotFoundException: #ejb.VehicleIF not found
		at com.sun.enterprise.naming.impl.TransientContext.doLookup(TransientContext.java:235)
		at com.sun.enterprise.naming.impl.TransientContext.lookup(TransientContext.java:203)
		at com.sun.enterprise.naming.impl.SerialContextProviderImpl.lookup(SerialContextProviderImpl.java:38)
		at com.sun.enterprise.naming.impl.RemoteSerialContextProviderImpl.lookup(RemoteSerialContextProviderImpl.java:92)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.base/java.lang.reflect.Method.invoke(Method.java:568)
		at com.sun.corba.ee.impl.presentation.rmi.ReflectiveTie.dispatchToMethod(ReflectiveTie.java:122)
		at com.sun.corba.ee.impl.presentation.rmi.ReflectiveTie._invoke(ReflectiveTie.java:152)
		at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatchToServant(ServerRequestDispatcherImpl.java:501)
		at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatch(ServerRequestDispatcherImpl.java:172)
		at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequestRequest(MessageMediatorImpl.java:1529)
		at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequest(MessageMediatorImpl.java:1405)
		at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleInput(MessageMediatorImpl.java:910)
		at com.sun.corba.ee.impl.protocol.giopmsgheaders.RequestMessage_1_2.callback(RequestMessage_1_2.java:192)
		at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequest(MessageMediatorImpl.java:674)
		at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.dispatch(MessageMediatorImpl.java:476)
		at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.doWork(MessageMediatorImpl.java:2202)
		at com.sun.corba.ee.impl.threadpool.ThreadPoolImpl$WorkerThread.performWork(ThreadPoolImpl.java:476)
		at com.sun.corba.ee.impl.threadpool.ThreadPoolImpl$WorkerThread.run(ThreadPoolImpl.java:519)
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 7.861 s - in client.AppClientTest
Stopping container using command: [/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home/bin/java, -jar, /private/tmp/platform-tck/glassfish-runner/jpa-platform-tck/target/glassfish8/glassfish/modules/admin-cli.jar, stop-domain, -t]
Waiting finished after 43 ms.
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  12.335 s
[INFO] Finished at: 2024-11-06T12:48:08-06:00
[INFO] ------------------------------------------------------------------------
starksm@Scotts-Mac-Studio platform-tck % 

```

So the appclient does not run a full CDI container. It is just doing custom injection of the `@EJB` references in the appclient main class. If I add such a reference, this tries to inject the ejb, but it ends up still using the same faulting JNDI binding and fails:

```
com.sun.enterprise.container.common.spi.util.InjectionException: Exception attempting to inject Remote ejb-ref name=ee.jakarta.tck.persistence.core.EntityGraph.Client/bean,Remote 3.x interface =com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF,ejb-link=Stateless3VehicleBean,lookup=,mappedName=,jndi-name=,refType=Session into class ee.jakarta.tck.persistence.core.EntityGraph.Client: Lookup failed for java:comp/env/ee.jakarta.tck.persistence.core.EntityGraph.Client/bean in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}]
        at com.sun.enterprise.container.common.impl.util.InjectionManagerImpl._inject(InjectionManagerImpl.java:530)
        at com.sun.enterprise.container.common.impl.util.InjectionManagerImpl.inject(InjectionManagerImpl.java:362)
        at com.sun.enterprise.container.common.impl.util.InjectionManagerImpl.injectClass(InjectionManagerImpl.java:165)
        at com.sun.enterprise.container.common.impl.util.InjectionManagerImpl.injectClass(InjectionManagerImpl.java:160)
        at org.glassfish.appclient.client.acc.AppClientContainer$ClientMainClassSetting.getClientMainClass(AppClientContainer.java:514)
        at org.glassfish.appclient.client.acc.AppClientContainer.getMainMethod(AppClientContainer.java:425)
        at org.glassfish.appclient.client.acc.AppClientContainer.completePreparation(AppClientContainer.java:335)
        at org.glassfish.appclient.client.acc.AppClientContainer.prepare(AppClientContainer.java:254)
        at org.glassfish.appclient.client.AppClientFacade.prepareACC(AppClientFacade.java:224)
        at org.glassfish.appclient.client.acc.agent.AppClientContainerAgent.premain(AppClientContainerAgent.java:60)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:491)
        at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:503)
Caused by: javax.naming.NamingException: Lookup failed for java:comp/env/ee.jakarta.tck.persistence.core.EntityGraph.Client/bean in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: No object bound for java:comp/env/ee.jakarta.tck.persistence.core.EntityGraph.Client/bean]
        at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:836)
        at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:269)
        at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:413)
        at com.sun.enterprise.naming.impl.GlassfishNamingManagerImpl.lookup(GlassfishNamingManagerImpl.java:565)
        at com.sun.enterprise.container.common.impl.util.InjectionManagerImpl._inject(InjectionManagerImpl.java:470)
        ... 15 more
Caused by: javax.naming.NameNotFoundException: No object bound for java:comp/env/ee.jakarta.tck.persistence.core.EntityGraph.Client/bean
        at com.sun.enterprise.naming.impl.JavaURLContext.lookup(JavaURLContext.java:135)
        at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:818)
        ... 19 more
        Suppressed: javax.naming.NamingException: Exception resolving Ejb for 'Remote ejb-ref name=ee.jakarta.tck.persistence.core.EntityGraph.Client/bean,Remote 3.x interface =com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF,ejb-link=Stateless3VehicleBean,lookup=,mappedName=,jndi-name=,refType=Session' .  Actual (possibly internal) Remote JNDI name used for lookup is '#com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF' [Root exception is javax.naming.NamingException: Lookup failed for #com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: #com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF not found]]
                at com.sun.ejb.EjbNamingReferenceManagerImpl.resolveEjbReference(EjbNamingReferenceManagerImpl.java:159)
                at com.sun.enterprise.container.common.impl.ComponentEnvManagerImpl$EjbReferenceProxy.create(ComponentEnvManagerImpl.java:975)
                at com.sun.enterprise.naming.impl.GlassfishNamingManagerImpl.lookup(GlassfishNamingManagerImpl.java:621)
                at com.sun.enterprise.naming.impl.GlassfishNamingManagerImpl.lookup(GlassfishNamingManagerImpl.java:587)
                at com.sun.enterprise.naming.impl.JavaURLContext.lambda$lookup$0(JavaURLContext.java:153)
                at com.sun.enterprise.naming.impl.JavaURLContext.lookupOrCollectException(JavaURLContext.java:550)
                at com.sun.enterprise.naming.impl.JavaURLContext.lookup(JavaURLContext.java:153)
                ... 20 more
        Caused by: javax.naming.NamingException: Lookup failed for #com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.url.pkgs=com.sun.enterprise.naming, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl}] [Root exception is javax.naming.NameNotFoundException: #com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF not found]
                at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:836)
                at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:257)
                at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
                at com.sun.ejb.EjbNamingReferenceManagerImpl.resolveEjbReference(EjbNamingReferenceManagerImpl.java:154)
                ... 26 more
        Caused by: javax.naming.NameNotFoundException: #com.sun.ts.tests.common.vehicle.stateless3.Stateless3VehicleIF not found
                at com.sun.enterprise.naming.impl.TransientContext.doLookup(TransientContext.java:234)
                at com.sun.enterprise.naming.impl.TransientContext.lookup(TransientContext.java:202)
                at com.sun.enterprise.naming.impl.SerialContextProviderImpl.lookup(SerialContextProviderImpl.java:37)
                at com.sun.enterprise.naming.impl.RemoteSerialContextProviderImpl.lookup(RemoteSerialContextProviderImpl.java:85)
                at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
                at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                at java.base/java.lang.reflect.Method.invoke(Method.java:568)
                at com.sun.corba.ee.impl.presentation.rmi.ReflectiveTie.dispatchToMethod(ReflectiveTie.java:122)
                at com.sun.corba.ee.impl.presentation.rmi.ReflectiveTie._invoke(ReflectiveTie.java:152)
                at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatchToServant(ServerRequestDispatcherImpl.java:501)
                at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatch(ServerRequestDispatcherImpl.java:172)
                at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequestRequest(MessageMediatorImpl.java:1529)
                at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequest(MessageMediatorImpl.java:1405)
                at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleInput(MessageMediatorImpl.java:910)
                at com.sun.corba.ee.impl.protocol.giopmsgheaders.RequestMessage_1_2.callback(RequestMessage_1_2.java:192)
                at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequest(MessageMediatorImpl.java:674)
                at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.dispatch(MessageMediatorImpl.java:476)
                at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.doWork(MessageMediatorImpl.java:2202)
                at com.sun.corba.ee.impl.threadpool.ThreadPoolImpl$WorkerThread.performWork(ThreadPoolImpl.java:476)
                at com.sun.corba.ee.impl.threadpool.ThreadPoolImpl$WorkerThread.run(ThreadPoolImpl.java:519)
```

I'm not clear on how this differs from the simple case that works, so I'll see if I can determine what differs.

--- a/appserver/common/glassfish-naming/src/main/java/com/sun/enterprise/naming/impl/JavaURLContext.java
+++ b/appserver/common/glassfish-naming/src/main/java/com/sun/enterprise/naming/impl/JavaURLContext.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2022 Contributors to the Eclipse Foundation
+ * Copyright (c) 2022, 2026 Contributors to the Eclipse Foundation
  * Copyright (c) 2006, 2018 Oracle and/or its affiliates. All rights reserved.
  *
  * This program and the accompanying materials are made available under the
@@ -180,6 +180,10 @@ public final class JavaURLContext implements Context, Cloneable {
                 return obj;
             }
         }
+        // All strategies failed. The collector carries the cause of each attempt as a suppressed
+        // exception; log them here so the reason is available even when callers only report the
+        // top-level "No object bound" message.
+        LOG.log(Level.DEBUG, () -> "No naming strategy could resolve '" + name + "'.", e);
         throw e;
     }
 

--- a/appserver/ejb/ejb-container/src/main/java/com/sun/ejb/EjbNamingReferenceManagerImpl.java
+++ b/appserver/ejb/ejb-container/src/main/java/com/sun/ejb/EjbNamingReferenceManagerImpl.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2022, 2025 Contributors to the Eclipse Foundation.
+ * Copyright (c) 2022, 2025, 2026 Contributors to the Eclipse Foundation.
  * Copyright (c) 1997, 2018 Oracle and/or its affiliates. All rights reserved.
  *
  * This program and the accompanying materials are made available under the
@@ -150,9 +150,19 @@ public class EjbNamingReferenceManagerImpl implements EjbNamingReferenceManager 
 
             } catch(Exception e) {
                 // Important to make the real underlying lookup name part of the exception.
-                NamingException ne = new NamingException("Exception resolving Ejb for '" +
-                    ejbRefDesc + "' .  Actual (possibly internal) Remote JNDI name used for lookup is '" +
-                    remoteJndiName + "'");
+                StringBuilder msg = new StringBuilder("Exception resolving Ejb for '").append(ejbRefDesc)
+                    .append("'.  Actual (possibly internal) Remote JNDI name used for lookup is '")
+                    .append(remoteJndiName).append('\'');
+                if (ejbRefDesc.isLinked() && !ejbRefDesc.hasJndiName() && !ejbRefDesc.hasLookupName()) {
+                    // With no jndi-name or lookup-name, the remote name above degenerates to just
+                    // the '#<interface>' suffix, which is never bound. Spell out the real cause
+                    // rather than leaving a cryptic NameNotFoundException for that name.
+                    msg.append(". The reference is resolved only by ejb-link '").append(ejbRefDesc.getLinkName())
+                        .append("', but that link was not mapped to a bound JNDI name. Add a lookup-name")
+                        .append(" (or jndi-name) to the reference, or make sure the linked bean is part of")
+                        .append(" the same application so the ejb-link can be resolved.");
+                }
+                NamingException ne = new NamingException(msg.toString());
                 ne.initCause(e);
                 throw ne;
             } finally {

